### PR TITLE
Added the Recursive Flag

### PR DESCRIPTION
**Pull request by KingKhay**

### CODE REVIEW

File: ls/ls.go [hunk 1]
```diff
@@ -54,6 +60,34 @@ func (l *LS) lsDotfiles() {
...
+
+		func(path string, info os.FileInfo, err error) error {
+
+			if err != nil {
+				return err
+			}
+
+			//func IsHiddenFile(filename string) (bool, error) {
+			//	return filename[0] == '.', nil
+			//}
```
Comment (profclems) [inline ls/ls.go]: Seems we're not using this since it's commented out. Please remove it

File: main.go [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package main
 
 import (
-	"os"
-
 	"github.com/pee2pee/lse/ls"
 	"github.com/spf13/cobra"
+	"os"
+	_ "path/filepath"
```
Comment (profclems) [inline main.go]: remove unused import `path/filepath`

File: ls/ls.go [hunk 3]
```diff
@@ -54,6 +60,34 @@ func (l *LS) lsDotfiles() {
...
+			if !isHiddenPath(path, l.A) {
+				fmt.Println(path)
+				return err
+			}
+			return nil
+		})
+
+	if err != nil {
+
+		log.Println(err)
```
Comment (profclems) [inline ls/ls.go]: `return err` instead of `log.Println(err)`

File: ls/ls.go [hunk 4]
```diff
@@ -54,6 +60,34 @@ func (l *LS) lsDotfiles() {
...
+
+			if err != nil {
+				return err
+			}
+
+			//func IsHiddenFile(filename string) (bool, error) {
+			//	return filename[0] == '.', nil
+			//}
+			if !isHiddenPath(path, l.A) {
+				fmt.Println(path)
```
Comment (profclems) [inline ls/ls.go]: use `fmt.Fprintln(l.Stdout, path)`

**Review by profclems — CHANGES_REQUESTED** [review]
Great work! Just some minor comments to fix

File: ls/ls.go (hunk 1 above)
Comment (KingKhay) [reply ls/ls.go]: Very well.

File: main.go (hunk 2 above)
Comment (KingKhay) [reply main.go]: Actually used the filepath's walk function to recursively print the subdirectories

File: ls/ls.go (hunk 3 above)
Comment (KingKhay) [reply ls/ls.go]: Fixed XOXO

File: main.go (hunk 2 above)
Comment (profclems) [reply main.go]: The `path/filepath` is used in `ls/ls.go`, not `main.go`, so no need to add it to main.go.

Comment (KingKhay) [reply main.go]: Sure. Fixed it:sunglasses:

**Review by profclems — APPROVED** [review]
LGTM

**Pull request merged**